src/mtype.c

```diff
@@ -5635,8 +5635,8 @@ Type *TypeFunction::semantic(Loc loc, Scope *sc)
                 }
                 else
                 {
-                    error(loc, "auto can only be used for template function parameters");
-                    errors = true;
+                    fparam->storageClass &= ~STCauto;
+                    fparam->storageClass |= STCrvref;
```

Comment:
From my experiment in #4729, this change might introduce an accepts-invalid bug - an invalid non-template function declaration `void foo(auto int x) {}` will be wrongly accepted?